apps/meteor/client/views/audit/components/SecurityLogsTable.tsx

```diff
@@ -0,0 +1,218 @@
+				timestamp={timestamp}
+				actor={actor}
+				setting={String(setting)}
+				changedFrom={changedFrom}
+				changedTo={changedTo}
+				onCancel={() => setModal(null)}
+			/>,
+		);
+	};
+
+	const getAudits = useEndpoint('GET', '/v1/audit.settings');
+
+	const { data, isLoading, isSuccess } = useQuery({
+		queryKey: ['audit.settings', query, itemsPerPage, current],
+
+		queryFn: async () => {
+			return getAudits({ ...query, ...(itemsPerPage && { count: itemsPerPage }), ...(current && { offset: current }) });
+		},
+	});
+
+	return (
+		<>
+			<Box mb={16} display='flex' flexWrap='nowrap' alignItems='flex-end'>
+				<Field flexShrink={1}>
+					<Margins inline={6}>
+						<FieldLabel>{t('Date')}</FieldLabel>
+						<DateRangePicker display='flex' flexGrow={1} value={dateRange} onChange={setDateRange} />
+					</Margins>
+				</Field>
+				<Field width={300}>
+					<Margins inline={6}>
+						<FieldLabel>{t('Setting')}</FieldLabel>
+						<SettingSelect value={setting} onChange={setSetting} />
+					</Margins>
+				</Field>
+				<ButtonGroup>
+					<Margins inline={6}>
+						<Button secondary onClick={handleClearFilters}>
+							{t('Clear_filters')}
+						</Button>
+						<Button primary onClick={handleApplyFilters}>
+							{t('Apply_filters')}
+						</Button>
+					</Margins>
+				</ButtonGroup>
+			</Box>
+
+			{isLoading && (
+				<GenericTable>
+					<GenericTableHeader>
+						<GenericTableHeaderCell>{t('Actor')}</GenericTableHeaderCell>
+						<GenericTableHeaderCell>{t('Timestamp')}</GenericTableHeaderCell>
+						<GenericTableHeaderCell>{t('Setting')}</GenericTableHeaderCell>
+						<GenericTableHeaderCell>{t('Changed_from')}</GenericTableHeaderCell>
+						<GenericTableHeaderCell>{t('Changed_to')}</GenericTableHeaderCell>
+					</GenericTableHeader>
+					<GenericTableBody>
+						<GenericTableLoadingRow cols={5} />
+					</GenericTableBody>
+				</GenericTable>
+			)}
+			{isSuccess && data.total === 0 && (
+				<GenericNoResults
+					title={t('No_results_found')}
+					description={t('Try_different_filters')}
+					buttonTitle={t('Clear_filters')}
+					buttonAction={handleClearFilters}
+				/>
+			)}
+			{isSuccess && data.total > 0 && (
+				<GenericTable striped>
+					<GenericTableHeader>
+						<GenericTableHeaderCell>{t('Actor')}</GenericTableHeaderCell>
+						<GenericTableHeaderCell>{t('Timestamp')}</GenericTableHeaderCell>
+						<GenericTableHeaderCell>{t('Setting')}</GenericTableHeaderCell>
+						<GenericTableHeaderCell>{t('Changed_from')}</GenericTableHeaderCell>
+						<GenericTableHeaderCell>{t('Changed_to')}</GenericTableHeaderCell>
+					</GenericTableHeader>
+					<GenericTableBody>
+						{data.events.map((item) => {
+							const setting = item.data.find((item) => item.key === 'id');
+							const previous = item.data.find((item) => item.key === 'previous');
+							const current = item.data.find((item) => item.key === 'current');
+							return (
+								<GenericTableRow
+									key={item.ts}
```

Comment:
Sometimes (rare, but does happen) the timestamp can be the same for multiple records. Can we use the record _id instead of ts here?